pkg/controller/certificates/sync.go

```diff
@@ -216,6 +219,7 @@ func (c *Controller) updateSecret(name, namespace string, cert, key []byte) (*ap
 			Data: map[string][]byte{},
 		}
 	}
+	secret.OwnerReferences = []metav1.OwnerReference{*metav1.NewControllerRef(crt, certificateGVK)}
```

Comment:
So I think we need to be very careful when setting this field.. Namely:

* If a resource already exists, but has no OwnerReferences, do we add our own or bail out early?
* If a resource already exists, but is 'owned' by a different resource, do we add our own & remove existing or bail out early?
* If a resource already exists, but is 'owned' by a different *Certificate* resource (this could indicate a user has `kubectl delete certificate xyz`, but GC didn't run, do we add our own and remove the existing or bail out early?

Importantly (1) and (2) are likely to come up quickly with users coming from k-c-m or kube-lego.